lib/StripeResource.js

```diff
@@ -1,619 +1,434 @@
-            );
-          }
-        });
-    };
-
-    const prepareAndMakeRequest = (error, data) => {
-      if (error) {
-        return callback(error);
-      }
-
-      requestData = data;
-
-      this._stripe.getClientUserAgent((clientUserAgent) => {
-        const apiVersion = this._stripe.getApiField('version');
-        const headers = this._makeHeaders(
-          auth,
-          requestData.length,
-          apiVersion,
-          clientUserAgent,
-          method,
-          options.headers,
-          options.settings
-        );
-
-        makeRequest(apiVersion, headers);
-      });
-    };
-
-    if (this.requestDataProcessor) {
-      this.requestDataProcessor(
-        method,
-        data,
-        options.headers,
-        prepareAndMakeRequest
-      );
-    } else {
-      prepareAndMakeRequest(null, utils.stringifyRequestData(data || {}));
-    }
-  },
+            const requestRetries = numRetries || 0;
+            const maxRetries = this._getMaxNetworkRetries(options.settings);
+            this._stripe._emitter.emit('request', requestEvent);
+            req
+                .then((res) => {
+                if (this._shouldRetry(res, requestRetries, maxRetries)) {
+                    return retryRequest(makeRequest, apiVersion, headers, requestRetries, res.getHeaders()['retry-after']);
+                }
+                else if (options.streaming && res.getStatusCode() < 400) {
+                    return this._streamingResponseHandler(requestEvent, callback)(res);
+                }
+                else {
+                    return this._jsonResponseHandler(requestEvent, callback)(res);
+                }
+            })
+                .catch((error) => {
+                if (this._shouldRetry(null, requestRetries, maxRetries, error)) {
+                    return retryRequest(makeRequest, apiVersion, headers, requestRetries, null);
+                }
+                else {
+                    const isTimeoutError = error.code && error.code === HttpClient.TIMEOUT_ERROR_CODE;
+                    return callback.call(this, new StripeConnectionError({
+                        message: isTimeoutError
+                            ? `Request aborted due to timeout being reached (${timeout}ms)`
+                            : this._generateConnectionErrorMessage(requestRetries),
+                        detail: error,
+                    }));
+                }
+            });
+        };
+        const prepareAndMakeRequest = (error, data) => {
+            if (error) {
+                return callback(error);
+            }
+            requestData = data;
+            this._stripe.getClientUserAgent((clientUserAgent) => {
+                const apiVersion = this._stripe.getApiField('version');
+                const headers = this._makeHeaders(auth, requestData.length, apiVersion, clientUserAgent, method, options.headers, options.settings);
+                makeRequest(apiVersion, headers, 0);
```

Comment:
Note to self: get back to that `0` at the end.

Reply:
Resolving for now, feel free to open this back up if you want me to change something!